python/grass/__init__.py

```diff
@@ -75,16 +93,48 @@ def _translate(text):
             # does not raise an exception even if the locale settings is broken
             # or the translation files were not installed.
             fallback = True
-            translation = gettext.translation(
-                "grasslibs", locale_dir, fallback=fallback
-            )
-            # Add other domains as fallback.
-            translation.add_fallback(
-                gettext.translation("grassmods", locale_dir, fallback=fallback)
-            )
-            translation.add_fallback(
-                gettext.translation("grasswxpy", locale_dir, fallback=fallback)
-            )
+            translation = None
+            domains = ("grasslibs", "grassmods", "grasswxpy")
+
+            def apply_trans(kwargs):
+                nonlocal translation
+                if kwargs["domain"] == "grasslibs":
+                    translation = gettext.translation(**kwargs)
+                else:
+                    translation.add_fallback(gettext.translation(**kwargs))
+
+            try:
+                # Get user defined lang
+                lang = _UserSettings.Get(
+                    group="language",
+                    key="locale",
+                    subkey="lc_all",
+                )
+            except NameError:
+                # If _UserSettings isn't imported during tests (without wxGUI)
+                lang = None
+
+            if lang and lang not in ("en", "system"):
+                for domain in domains:
+                    apply_trans(
+                        {
+                            "domain": domain,
+                            "localedir": locale_dir,
+                            "fallback": fallback,
+                            "languages": (lang,),
+                        }
+                    )
+            elif lang == "en":
+                translation = gettext.NullTranslations()
+            else:
```

Comment:
This is meant for system, right? Likely the same call with language/lang being set to None will remove the need for another loop in the code (see the comment for the first loop).